File: AlgorithmsCSharp/src/AlgorithmsCSharp/Sorting/SelectionSort.cs
using AlgorithmsCSharp.Collections;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AlgorithmsCSharp.Sorting
{
    public class SelectionSort : ISortAlgorithm
    {
        public void Sort(IComparable[] items)
        {
            int length = items.Length;
            int min = 0;
            for (int i = 0; i < length; i++)
            {
                min = i;
                for (int j = i + 1; j < length; j++)
                {
                    if (ArrayHelper.IsLess(items[j], items[min]))
                    {
                        min = j;
                    }
                }

                ArrayHelper.Swap(items, i, min);
            }
        }
    }
}

File: AlgorithmsJava/src/main/java/LearnedAlgorithms/SelectionSort.java
package LearnedAlgorithms;

public class SelectionSort {
    public void sort(String[] items) {
        int i = 0;
        for (i = 0; i < items.length; i++) {
            String min = items[i];
            int minIndex = i;
            for (int j = i + 1; j < items.length; j++) {
                min = items[minIndex];
                String current = items[j];
                if (current.compareTo(min) < 0) {
                    minIndex = j;
                }
            }
            String temp = items[i];
            items[i] = items[minIndex];
            items[minIndex] = temp;
        }
    }
}
